src/server/modes/charts/plugins/datalens/preparers/line/helpers/line-mappers/prepareLines.ts

```diff
@@ -359,6 +363,9 @@ export const prepareLines = (args: PrepareLinesArgs) => {
             };
         }
 
+        // id set so that filtering by legend could work as expected
+        lines[keys.key].id = currentLine.legendTitle || currentLine.title;
```

Comment:
It looks like the code line above. 
Can we remove one of the lines? Either add it to the segment or remove from other part